anchor/eth/src/sync.rs

```diff
@@ -0,0 +1,248 @@
+        // Construct HTTP Provider
+        let http_url = "dummy_http".parse().unwrap(); // TODO!(), get this from config
+        let rpc_client: Arc<RpcClient> = Arc::new(ProviderBuilder::new().on_http(http_url));
+
+        // Experiment with retry clients for both websocket and http
+        /* 
+        let client = ClientBuilder::default()
+            .layer(RetryBackoffLayer::new(10, 300, 300))
+            .http(http_url);
+        let retry_rpc_client = ProviderBuilder::new().on_client(client);
+        */
+
+        // Construct Websocket Provider
+        let ws_url = "dummy ws"; // TODO!(), get this from config
+        let ws_client = ProviderBuilder::new()
+            .on_ws(WsConnect::new(ws_url))
+            .await
+            .map_err(|e| format!("Failed to bind to WS: {}, {}", ws_url, e))?;
+
+        // Pass db access here
+        let event_processor = EventProcessor::new();
+
+        Ok(Self {
+            rpc_client,
+            ws_client,
+            event_processor,
+        })
+    }
+
+    // Top level function to sync data
+    pub async fn sync(&self) -> Result<(), String> {
+        // first, perform a historical sync
+        self.historical_sync().await?;
+
+        // start the live sync, options
+        // 1) spawn the sync off in its own long running task and return
+        // 2) transition into live sync and signal AtomicBool to coordinator
+        self.live_sync().await?;
+        todo!()
+    }
+
+    /// Perform a historical sync from the contract deployment block to catch up to the current
+    /// state of the SSV network
+    async fn historical_sync(&self) -> Result<(), String> {
+        // todo!(), differential between fresh sync and when we have already synced up to some block
+        let mut start_block = CONTRACT_DEPLOYMENT_BLOCK;
+        loop {
+            // get the current block and make sure we have blocks to sync
+            let current_block = self
+                .rpc_client
+                .get_block_number()
+                .await
+                .map_err(|e| format!("Unable to fetch block number {}", e))?;
+            if current_block < FOLLOW_DISTANCE {
+                break;
+            }
+
+            // calculate end block w/ follow distance
+            let end_block = current_block - FOLLOW_DISTANCE;
+            if end_block < start_block {
+                break;
+            }
+
+            // Chunk the start and end block range into a set of ranges of size BATCH_SIZE
+            // and construct a future to fetch the logs in each range
+            let tasks: Vec<_> = (start_block..=current_block)
+                .step_by(BATCH_SIZE as usize)
+                .map(|start| {
+                    let (start, end) =
+                        (start, std::cmp::min(start + BATCH_SIZE - 1, current_block));
+                    self.fetch_logs(start, end)
+                })
+                .collect();
```

Comment:
Potentially yes. I have not confirmed yet but I do not believe the current total amount of events emitted by the contract is very much. Although, as the network gains popularity and continues on over the years this may be a problem so it is probably best to make this change. Thanks!